src/test/java/seedu/address/model/transaction/TransactionIdTest.java

```diff
@@ -0,0 +1,74 @@
+package seedu.address.model.transaction;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import java.util.UUID;
+
+import org.junit.jupiter.api.Test;
+
+public class TransactionIdTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new TransactionId(null));
+    }
+
+    @Test
+    public void constructor_invalidTransactionId_throwsIllegalArgumentException() {
+        String invalidTransactionId = "";
+        assertThrows(IllegalArgumentException.class, () -> new TransactionId(invalidTransactionId));
+    }
+
+    @Test
+    public void getTransactionIdTest() {
+        UUID uuid = UUID.randomUUID();
+        TransactionId transactionId = new TransactionId(uuid.toString());
+        assertEquals(transactionId.getTransactionId(), uuid.toString());
+
+        UUID otherId = UUID.randomUUID();
+        assertNotEquals(transactionId.getTransactionId(), otherId.toString());
+    }
+
+    @Test
+    public void isValidTransactionId() {
+        // null transaction id
+        assertThrows(NullPointerException.class, () -> TransactionId.isValidTransactionId(null));
+
+        // invalid transaction id
+        assertFalse(TransactionId.isValidTransactionId("")); // empty string
+        assertFalse(TransactionId.isValidTransactionId(" ")); // spaces only
+        assertFalse(TransactionId.isValidTransactionId("3")); // only 3 numbers
+        assertFalse(TransactionId.isValidTransactionId("transaction")); // alphabets that greater than f
+        assertFalse(TransactionId.isValidTransactionId("9312 1534")); // spaces within digits
+
+        // valid transaction id
+        assertTrue(TransactionId.isValidTransactionId("cbd83eae-1e34-4c6e-90c1-e9fbc3d0e3ef"));
+        assertTrue(TransactionId.isValidTransactionId("af43a9b7-f896-44ad-be92-39ae801462d8"));
+        assertTrue(TransactionId.isValidTransactionId("1b505667-4970-47c2-8484-34a9b43ce260"));
```

Comment:
Does this work for capital letters A-F?

Reply:
yes, as indicated in the Transaction Id class's regex.